Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,177 @@
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func checkSyncStatus() {
+        let accountsFilterFn: (DcrlibwalletAccount) -> Bool = { $0.totalBalance > 0 || $0.name != "imported" }
+        guard let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0, accountsFilterFn: accountsFilterFn) }).first,
+            let account = wallet.accounts.first else {
+                self.moreMenuButton.isEnabled = false
+                self.mainContentViewHolder.isHidden = true
+                self.syncInProgressLabel.isHidden = false
+                return
+        }
+
+        self.updateSelectedAccount(wallet.id, account)
+
+        mainContentViewHolder.isHidden = false
+        syncInProgressLabel.isHidden = true
+        self.moreMenuButton.isEnabled = true
+    }
+
+    private func displayAddressAndQRCode(receiveAddress: String) {
+        DispatchQueue.main.async {
+            self.walletAddressLabel.text = receiveAddress
+            self.addressQRCodeImageView.image = self.generateQRCodeImage(
+                for: receiveAddress
+            )
+        }
+    }
+
+    private func generateNewAddress() {
+        self.oldAddress = self.walletAddressLabel.text!
+        self.getNextAddress()
+    }
+
+    @objc private func getNextAddress() {
+        if let wallet = self.selectedWallet,
+            let account = self.selectedAccount {
+                let receiveAddress = wallet.nextAddress(account.number, error: nil)
+                if self.oldAddress != receiveAddress {
+                    self.displayAddressAndQRCode(receiveAddress: receiveAddress)
+                } else if receiveAddress != "" {
+                    self.getNextAddress()
+                }
+        }
+    }
+
+    private func generateQRCodeImage(for address: String) -> UIImage? {
+        guard let colorFilter = CIFilter(name: "CIFalseColor") else { return nil }
+        let filter = CIFilter(name: "CIQRCodeGenerator")
+        filter?.setValue(address.utf8Bits, forKey: "inputMessage")
+
+        colorFilter.setDefaults()
+        colorFilter.setValue(filter!.outputImage, forKey: "inputImage")
+        colorFilter.setValue(CIColor.black, forKey: "inputColor0")
+        colorFilter.setValue(CIColor.clear, forKey: "inputColor1")
+
+        if let qrImage = colorFilter.outputImage {
+            let frame = self.addressQRCodeImageView.frame
+            let smallerSide = frame.size.width < frame.size.height ? frame.size.width : frame.size.height
+            let scale = smallerSide/qrImage.extent.size.width
+            let transformedImage = qrImage.transformed(by: CGAffineTransform(scaleX: scale, y: scale))
+            return UIImage(ciImage: transformedImage)
```

Comment:
In line with the previous comment,
```suggestion
            self.addressQRCodeImageView.image = UIImage(ciImage: transformedImage)
```